lib/nylas/http_client.rb

```diff
@@ -178,10 +178,15 @@ def handle_failed_response(result:, response:)
 
     def handle_anticipated_failure_mode(http_code:, response:)
       return if http_code == 200
-      return unless response.is_a?(Hash)
 
       exception = HTTP_CODE_TO_EXCEPTIONS.fetch(http_code, APIError)
-      raise exception.new(response[:type], response[:message], response.fetch(:server_error, nil))
+      response = parse_response(response)
```

Comment:
Looks like you are overwriting this variable? What do you think about using a new name?

Reply:
Yes. Done here 3a3ea3a305e30f7a869daebf7affe4ba9a06207a